event/parsing/event.go

```diff
@@ -46,81 +61,75 @@ type BootTimeCalc struct {
 	Auth    acquire.Acquirer
 }
 
-var destinationRegex = regexp.MustCompile(`^(?P<event>[^\\/]+)\\/((?P<prefix>(?i)mac|uuid|dns|serial):(?P<id>[^\\/]+))\\/(?P<type>[^\\/\\s]+)`)
+var destinationRegex = regexp.MustCompile(`^(?P<event>[^\/]+)\/((?P<prefix>(?i)mac|uuid|dns|serial):(?P<id>[^\/]+))\/(?P<type>[^\/\s]+)`)
 var onlineRegex = regexp.MustCompile(".*/online$")
 var offlineRegex = regexp.MustCompile(".*/offline$")
 
-func (b BootTimeCalc) Parse(msg wrp.Message) error {
-	/**
-	Steps to calculate boot time:
+/* Parse calculates boot time of devices by querying codex for the latest offline events and performing
+calculations. An analysis of codex events is only triggered by a device online event from caduceus.
+Steps to calculate boot time:
 	1) Determine if message is online event
 	2) Get lastest Offline event from Codex where metadata field of /boot-time differs of online event.
 	3) Subtract Online birthdate from steps 2 event Birthdate
 	4) Record Metric
-	*/
+*/
+func (b BootTimeCalc) Parse(msg wrp.Message) error {
+	// if event is not an online event, do not continue with calculations
 	if !destinationRegex.MatchString(msg.Destination) || !onlineRegex.MatchString(msg.Destination) {
-		return errors.New("event is not online")
+		logging.Debug(b.Logger).Log(logging.MessageKey(), "event is not an online event")
```

Comment:
Discussed this with @kcajmagic and it's possible for glaukos to be registered for all events, not just online events. In that case, does adding a metric for this tell us anything useful? It's not an error if we've configured glaukos to subscribe to events that include non-online events and we hit this line, since we pass all the events we receive through all of the glaukos parsers.

Reply:
Sounds good! 👍